libsrc/leddevice/dev_net/LedDevicePhilipsHue.cpp

```diff
@@ -126,95 +122,134 @@ float CiColor::getDistanceBetweenTwoPoints(CiColor p1, CiColor p2)
 	return sqrt(dx * dx + dy * dy);
 }
 
-QByteArray PhilipsHueBridge::get(QString route)
+PhilipsHueBridge::PhilipsHueBridge(Logger* log, QString host, QString username)
+	: QObject()
+	, log(log)
+	, host(host)
+	, username(username)
+{
+	// setup reconnection timer
+	bTimer.setInterval(5000);
+	bTimer.setSingleShot(true);
+
+	connect(&bTimer, &QTimer::timeout, this, &PhilipsHueBridge::bConnect);
+	connect(&manager, &QNetworkAccessManager::finished, this, &PhilipsHueBridge::resolveReply);
+}
+
+void PhilipsHueBridge::bConnect(void)
+{
+	if(username.isEmpty() || host.isEmpty())
+	{
+		Error(log,"Username or IP Address is empty!");
+	}
+	else
+	{
+		QString url = QString("http://%1/api/%2").arg(host).arg(username);
+		Debug(log, "Connect to bridge %s", QSTRING_CSTR(url));
+
+		QNetworkRequest request(url);
+		manager.get(request);
+	}
+}
+void PhilipsHueBridge::resolveReply(QNetworkReply* reply)
 {
-	QString url = QString("http://%1/api/%2/%3").arg(host).arg(username).arg(route);
-	Debug(log, "Get %s", url.toStdString().c_str());
-	// Perfrom request
-	QNetworkRequest request(url);
-	QNetworkReply* reply = manager->get(request);
-	// Connect requestFinished signal to quit slot of the loop.
-	QEventLoop loop;
-	loop.connect(reply, SIGNAL(finished()), SLOT(quit()));
-	// Go into the loop until the request is finished.
-	loop.exec();
-	// Read all data of the response.
-	QByteArray response = reply->readAll();
-	// Free space.
+	// TODO use put request also for network error checking with decent threshold
+	if(reply->operation() == QNetworkAccessManager::GetOperation)
+	{
+		if(reply->error() == QNetworkReply::NoError)
+		{
+			QByteArray response = reply->readAll();
+			QJsonParseError error;
+			QJsonDocument doc = QJsonDocument::fromJson(response, &error);
+			if (error.error != QJsonParseError::NoError)
+			{
+				Error(log, "Got invalid response from bridge");
+				return;
+			}
+			// check for authorization
+			if(doc.isArray())
+			{
+				Error(log, "Authorization failed, username invalid");
+				return;
+			}
+			else
```

Comment:
Code lile
If ()
{
 return
}
else
{
action
}

Can be written as:
If ()
{
Return
}

action

No need fot 'else' branch

Reply:
Of course, now where you name it